libs/dh/reports/feature-measurement-reports/src/components/request-report/request-as-modal.component.ts

```diff
@@ -0,0 +1,135 @@
+//#region License
+/**
+ * @license
+ * Copyright 2020 Energinet DataHub A/S
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License2");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+//#endregion
+import { Component, DestroyRef, inject, signal } from '@angular/core';
+import {
+  FormControl,
+  FormGroup,
+  NonNullableFormBuilder,
+  ReactiveFormsModule,
+  Validators,
+} from '@angular/forms';
+import { RxPush } from '@rx-angular/template/push';
+import { TranslocoDirective } from '@jsverse/transloco';
+import { Apollo } from 'apollo-angular';
+import { takeUntilDestroyed } from '@angular/core/rxjs-interop';
+
+import { WattButtonComponent } from '@energinet-datahub/watt/button';
+import { WattDropdownComponent } from '@energinet-datahub/watt/dropdown';
+import { WATT_MODAL, WattModalService, WattTypedModal } from '@energinet-datahub/watt/modal';
+import { VaterStackComponent } from '@energinet-datahub/watt/vater';
+import { getActorOptions } from '@energinet-datahub/dh/shared/data-access-graphql';
+import { EicFunction, GetActorByIdDocument } from '@energinet-datahub/dh/shared/domain/graphql';
+import {
+  DhActorStorage,
+  PermissionService,
+} from '@energinet-datahub/dh/shared/feature-authorization';
+
+import { DhRequestReportModal } from './request-report-modal.component';
+
+type DhFormType = FormGroup<{
+  actorId: FormControl<string>;
+}>;
+
+@Component({
+  selector: 'dh-request-as-modal',
+  imports: [
+    RxPush,
+    ReactiveFormsModule,
+    TranslocoDirective,
+
+    WATT_MODAL,
+    VaterStackComponent,
+    WattDropdownComponent,
+    WattButtonComponent,
+  ],
+  templateUrl: './request-as-modal.component.html',
+})
+// eslint-disable-next-line @angular-eslint/component-class-suffix
+export class DhRequestAsModal extends WattTypedModal {
+  private readonly formBuilder = inject(NonNullableFormBuilder);
+  private readonly modalService = inject(WattModalService);
+  private readonly destroyRef = inject(DestroyRef);
+  private readonly apollo = inject(Apollo);
+
+  private readonly permissionService = inject(PermissionService);
+  private readonly actorStorage = inject(DhActorStorage);
+
+  form: DhFormType = this.formBuilder.group({
+    actorId: new FormControl<string>(this.actorStorage.getSelectedActorId(), {
+      validators: Validators.required,
+      nonNullable: true,
+    }),
+  });
+
+  actorOptions$ = getActorOptions(
```

Comment:
Maybe make a signal version of getActorOptions?

Reply:
Should be done now.